hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -498,72 +535,138 @@ private List<ContainerBlockMetadata> getBlocks(
+            index++;
+          }
+          if (index < notSCMContainers.size()) {
+            notSCMContainers = notSCMContainers.subList(index,
+                Math.min(index + limit, notSCMContainers.size()));
+          } else {
+            notSCMContainers = Collections.emptyList();
+          }
+        } else {
+          notSCMContainers = notSCMContainers.subList(0,
+              Math.min(limit, notSCMContainers.size()));
+        }
 
-      notSCMContainers.forEach(nonSCMContainer -> {
-        ContainerDiscrepancyInfo containerDiscrepancyInfo =
-            new ContainerDiscrepancyInfo();
-        containerDiscrepancyInfo.setContainerID(nonSCMContainer.getKey());
-        containerDiscrepancyInfo.setNumberOfKeys(
-            nonSCMContainer.getValue().getNumberOfKeys());
-        containerDiscrepancyInfo.setPipelines(nonSCMContainer.getValue()
-            .getPipelines());
-        containerDiscrepancyInfo.setExistsAt("OM");
-        containerDiscrepancyInfoList.add(containerDiscrepancyInfo);
-      });
+        notSCMContainers.forEach(nonSCMContainer -> {
+          ContainerDiscrepancyInfo containerDiscrepancyInfo =
+              new ContainerDiscrepancyInfo();
+          containerDiscrepancyInfo.setContainerID(nonSCMContainer.getKey());
+          containerDiscrepancyInfo.setNumberOfKeys(
+              nonSCMContainer.getValue().getNumberOfKeys());
+          containerDiscrepancyInfo.setPipelines(
+              nonSCMContainer.getValue().getPipelines());
+          containerDiscrepancyInfo.setExistsAt("OM");
+          containerDiscrepancyInfoList.add(containerDiscrepancyInfo);
+        });
+        break;
 
-      // Filter list of container Ids which are present in SCM but not in OM.
-      List<Long> nonOMContainers = scmNonDeletedContainers.stream()
-          .filter(containerId -> !omContainers.containsKey(containerId))
-          .collect(Collectors.toList());
+      case OM:
+        List<Long> nonOMContainers = scmNonDeletedContainers.stream()
+            .filter(containerId -> !omContainers.containsKey(containerId))
+            .collect(Collectors.toList());
 
-      List<Pipeline> pipelines = new ArrayList<>();
-      nonOMContainers.forEach(nonOMContainerId -> {
-        ContainerDiscrepancyInfo containerDiscrepancyInfo =
-            new ContainerDiscrepancyInfo();
-        containerDiscrepancyInfo.setContainerID(nonOMContainerId);
-        containerDiscrepancyInfo.setNumberOfKeys(0);
-        PipelineID pipelineID = null;
-        try {
-          pipelineID = containerManager.getContainer(
-                  ContainerID.valueOf(nonOMContainerId))
-              .getPipelineID();
-
-          if (null != pipelineID) {
-            pipelines.add(pipelineManager.getPipeline(pipelineID));
+        if (prevKey > 0) {
+          int index = 0;
+          while (index < nonOMContainers.size() &&
+              nonOMContainers.get(index) <= prevKey) {
```

Comment:
check the order of nonOmContainers, if ordered. If not will have incorrect output.

Reply:
The source of information for `nonOMContainers` is `containerManager.getContainers()`, which returns a list of containers based on ascending container IDs. As a result, nonOMContainers will also be in ascending order based on the container IDs.

Reply:
### Also tested it out with real time data of 36 containers 
<img width="1385" alt="image" src="https://github.com/apache/ozone/assets/98023601/4ca13464-5c92-428f-819e-bc2edccfb1b8">